### PR TITLE
Maxus Update

### PR DESCRIPTION
**Pull request by KilveEng**
individual cell volts and call temps added to metrics
polling rework
PID / Metrics rework

### CODE REVIEW

File: vehicle/OVMS.V3/components/vehicle_maxus_edeliver3/src/vehicle_med3.cpp [hunk 1]
```diff
@@ -29,117 +31,176 @@
...
-        memset(m_vin, 0, sizeof(m_vin));
-      
+
+        // Init CAN:
         RegisterCanBus(1,CAN_MODE_ACTIVE,CAN_SPEED_500KBPS);
-        PollSetPidList(m_can1,obdii_polls);
-        PollSetState(0);
+
+        // Init BMS:
+        BmsSetCellArrangementVoltage(96, 12);
```
Comment (dexterbg) [inline vehicle/OVMS.V3/components/vehicle_maxus_edeliver3/src/vehicle_med3.cpp:87]: As you've got 16 modules, you should replace 12 by 16 here

File: vehicle/OVMS.V3/components/vehicle_maxus_edeliver3/src/vehicle_med3.cpp [hunk 2]
```diff
@@ -29,117 +31,176 @@
...
+  switch (pid)
+  {
+      case vcuvin:  // VIN
+          strncat(m_vin,(char*)data,length);
+          if (mlremain==0)
+          {
+              StandardMetrics.ms_v_vin->SetValue(m_vin);
+              m_vin[0] = 0;
+          }
+          break;
```
Comment (dexterbg) [inline vehicle/OVMS.V3/components/vehicle_maxus_edeliver3/src/vehicle_med3.cpp:148]: You now have an assembled VIN in m_rxbuf, the assembly by strncat() isn't needed and won't work any longer. You only need to do `StandardMetrics.ms_v_vin->SetValue(m_rxbuf);` here. You also no longer need the `m_vin` buffer.

File: vehicle/OVMS.V3/components/vehicle_maxus_edeliver3/src/vehicle_med3.h [hunk 3]
```diff
@@ -44,27 +44,24 @@ class OvmsVehicleMaxed3 : public OvmsVehicle
...
   private:
+      void IncomingFrameCan1(CAN_frame_t* p_frame);
       void IncomingPollFrame(CAN_frame_t* frame);
-   
-
       void HandleVinMessage(uint8_t* data, uint8_t length, uint16_t remain);
-      void IncomingPollReply(canbus* bus, uint16_t type, uint16_t pid, uint8_t* data, uint8_t length, uint16_t mlremain);
-//      void IncomingPollBms(canbus* bus, uint16_t type, uint16_t pid, uint8_t* data, uint8_t length, uint16_t mlremain);
       /// A temporary store for the VIN
       char m_vin[17];
```
Comment (dexterbg) [inline vehicle/OVMS.V3/components/vehicle_maxus_edeliver3/src/vehicle_med3.h:61]: See above, no longer needed.

**Review by dexterbg — COMMENTED** [review]
I don't know what you want to remove from `changes.txt`, seems OK for me.

Do you want to work on my comments or shall I merge this up to here?

Regards,
Michael

**Comment by KilveEng** [comment]
Ill have a go at updating ;-)

**Pull request merged**